geodiff/src/drivers/sqlitedriver.cpp

```diff
@@ -49,20 +49,32 @@ class Sqlite3DbMutexLocker
 class Sqlite3SavepointTransaction
 {
   public:
-    explicit Sqlite3SavepointTransaction( std::shared_ptr<Sqlite3Db> db )
-      : mDb( db )
+    explicit Sqlite3SavepointTransaction( std::shared_ptr<Sqlite3Db> db, const Context *context )
+      : mDb( db ), mContext( context )
     {
       if ( sqlite3_exec( mDb.get()->get(), "SAVEPOINT changeset_apply", 0, 0, 0 ) != SQLITE_OK )
+      {
+        std::string errMsg = sqliteErrorMessage( mDb.get()->get() );
+        mContext->logger().error( errMsg );
         throw GeoDiffException( "Unable to start savepoint" );
+      }
     }
 
     ~Sqlite3SavepointTransaction()
     {
       if ( mDb )
       {
         // we had some problems - roll back any pending changes
-        sqlite3_exec( mDb.get()->get(), "ROLLBACK TO changeset_apply", 0, 0, 0 );
-        sqlite3_exec( mDb.get()->get(), "RELEASE changeset_apply", 0, 0, 0 );
+        if ( sqlite3_exec( mDb.get()->get(), "ROLLBACK TO changeset_apply", 0, 0, 0 ) != SQLITE_OK )
+        {
+          std::string errMsg = sqliteErrorMessage( mDb.get()->get() );
+          mContext->logger().error( errMsg );
```

Comment:
It would be good to reuse the error code, I am wondering for example is we cannot create function in logger:

`logger::error(sqlite3db db, std::string functionName);` or maybe create "our" function `sqlite3Exec` that  
runs `sqlite3_exec` and also reports problems if there are any.

Reply:
But we also try to catch errors which happen in other cases too, no? I mean not only during `sqlite3_exec` calls but also in other SQLite calls, so we can't completely get rid of this code. If I understand it correctly, we already have "our" function to execute queries and it now also has logging https://github.com/MerginMaps/geodiff/pull/173/files#diff-a50a1333bc366f1a8df64b2b7e7945e98a0b584045f22f92cc40169f283e2359R59-R66

Reply:
yeah I think any sqlite3 call can trigger something we need to call... you are right